src/features/courses/course-controller.ts

```diff
@@ -121,36 +121,86 @@ class CourseController {
         
         if (isOpen) {
             const date = new Date();
-            where[Sequelize.Op.and] = [
-                {
-                    [Sequelize.Op.or]: [
-                        {
-                            startDate: {
-                                [Sequelize.Op.lte]: date
-                            }
-                        },
-                        {
-                            [`$studentTopicOverride.${StudentTopicOverride.rawAttributes.startDate.field}$`]: {
-                                [Sequelize.Op.lte]: date
-                            }
-                        }
-                    ]
-                },
-                {
-                    [Sequelize.Op.or]: [
-                        {
-                            deadDate: {
-                                [Sequelize.Op.gte]: date
+            // If no userId is passed, show all active topics and topics with extensions (professor view)
+            // TODO: Consider breaking these complex queries into functions that describe their utility.
+            if (_.isNil(userId)) {
+                where[Sequelize.Op.or] = [
+                    {
+                        [Sequelize.Op.and]: [
+                            {
+                                startDate: {
+                                    [Sequelize.Op.lte]: date
+                                }
+                            },
+                            {
+                                deadDate: {
+                                    [Sequelize.Op.gte]: date
+                                }
+                            },
+                        ]
+                    },
+                    {
+                        [Sequelize.Op.and]: [
+                            {
+                                [`$studentTopicOverride.${StudentTopicOverride.rawAttributes.startDate.field}$`]: {
+                                    [Sequelize.Op.lte]: date,
+                                }
+                            },
+                            {
+                                [`$studentTopicOverride.${StudentTopicOverride.rawAttributes.deadDate.field}$`]: {
+                                    [Sequelize.Op.gte]: date,
+                                }
                             }
-                        },
-                        {
-                            [`$studentTopicOverride.${StudentTopicOverride.rawAttributes.deadDate.field}$`]: {
-                                [Sequelize.Op.gte]: date
+                        ]
+                    }
+                ];
+            } else {
+                // If you have overrides, use the overrides, else use the base daterange (student view)
```

Comment:
We never actually added the userId to the query, luckily we have the frontend filter